style_guides.md

```diff
@@ -0,0 +1,88 @@
+# Coding Styles and Guidelines
+**One Branch Per Feature**<br>
+When implementing different features for the project, a new branch should be created
+for each feature. When those features are finished, a pull request should be filed to
+merge that branch into the main branch. Although there are no guidelines for how to
+structure commit messages, do try to keep them as descriptive as possible.
+
+**Regarding Variable Names**<br>
+All variable names should be written in snake case while class member variables should
+consist of the "m_" prefix. However, since variables in structs are generally exposed
+outside of the class, it is fine to exclude the "m_" prefix. In snake case, all letters
+are in lower case and words are split by an underscore. See examples below:
+
+```C++
+class MyClass
+{
+public:
+	MyClass(int set_var)
+		: m_my_var(set_var) {};
+
+private:
+	int m_my_var;
+};
+
+struct MyStruct
+{
+	int* buffer;
+	int size;
+}
+```
+
+**Header Files and Source Files**<br>
+All header files (.h) and source files (.cpp | .c) will be placed in different folders.
+Header files should all go in the ```include``` folder while source files should go in
+the ```src``` folder. As the code base gets larger and larger, some source files will
+inevitably be grouped into different sub-directories within the ```src``` folder. In
+this case, the subdirectories in the "include" folder should be similar to that of
+the ```src``` folder. See examples below:
+
+```
+include
+	file1.h
+	file2.h
+	my_folder
+		file3.h
```

Comment:
Good example, please change the header extension to `.hpp` here as well to match with the `.cpp`.